include/swift/AST/Types.h

```diff
@@ -1246,6 +1251,10 @@ class alignas(1 << TypeAlignInBits) TypeBase
   /// Whether this is the AnyObject type.
   bool isAnyObject();
 
+  /// Return true if this type will never be an AnyObject existential after
+  /// substitution.
+  bool isNotPotentiallyAnyObject();
```

Comment:
nitpick: IMO it's better to avoid the negation in the function name -> `isPotentiallyAnyObject`.